packages/app/background-jobs-common/src/background-job-processor/spy/BackgroundJobProcessorSpy.ts

```diff
@@ -21,20 +23,26 @@ type SpyPromise<JobData extends object, JobReturn> = {
 export class BackgroundJobProcessorSpy<JobData extends object, JobReturn>
 	implements BackgroundJobProcessorSpyInterface<JobData, JobReturn>
 {
+	private readonly scheduledJobs: Map<string, SafeJob<JobData>>
+
 	private readonly jobProcessingResults: Map<string, JobProcessingResult<JobData, JobReturn>>
-	private promises: SpyPromise<JobData, JobReturn>[]
+	private finishedJobsPromises: SpyPromise<JobData, JobReturn>[]
+	private scheduledJobsPromises: SpyPromise<JobData, JobData>[]
 
 	constructor() {
 		this.jobProcessingResults = new Map()
-		this.promises = []
+		this.scheduledJobs = new Map()
+		this.finishedJobsPromises = []
+		this.scheduledJobsPromises = []
 	}
 
 	clear(): void {
 		this.jobProcessingResults.clear()
-		this.promises = []
+		this.scheduledJobs.clear()
+		this.finishedJobsPromises = []
 	}
 
-	waitForJobWithId(
+	waitForFinishedJobWithId(
```

Comment:
🟡 Not sure about this renaming, is it not possible to keep the same name and add a new status `scheduled` instead of a new method?